ingestion/src/metadata/ingestion/connections/test_connections.py

```diff
@@ -116,3 +108,15 @@ def test_connection_db_common(connection: Engine, steps=None) -> TestConnectionR
         raise SourceConnectionException(msg) from exc
 
     return None
+
+
+@timeout(seconds=120)
+def test_connection_db_common(connection: Engine, steps=None) -> TestConnectionResult:
+    """
+    Default implementation is the engine to test.
+
+    Test that we can connect to the source using the given engine
+    :param connection: Engine to test
+    :return: None or raise an exception if we cannot connect
+    """
+    return test_connection_engine(connection, steps)
```

Comment:
since you already created this wrapper for the time out, why not pass the timeout here as a parameter in `test_connection_db_common` and apply it for all `test_connection_engine`?